test/EspressoTEEVerifier.t.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "forge-std/Test.sol";
+import {EspressoTEEVerifier} from "../src/EspressoTEEVerifier.sol";
+import {IEspressoTEEVerifier} from "../src/interface/IEspressoTEEVerifier.sol";
+import {EspressoSGXTEEVerifier} from "../src/EspressoSGXTEEVerifier.sol";
+import {IEspressoSGXTEEVerifier} from "../src/interface/IEspressoSGXTEEVerifier.sol";
+
+contract EspressoTEEVerifierTest is Test {
+    address adminTEE = address(141);
+    address fakeAddress = address(145);
+
+    EspressoTEEVerifier espressoTEEVerifier;
+    EspressoSGXTEEVerifier espressoSGXTEEVerifier;
+    bytes32 enclaveHash =
+        bytes32(0x01f7290cb6bbaa427eca3daeb25eecccb87c4b61259b1ae2125182c4d77169c0);
+    bytes32 enclaveSigner =
+        bytes32(0x5fc862cb2e7e1f449f36a18b18aca08c20feaed0d411247816c281d596420cbb);
+    //  Address of the automata V3QuoteVerifier deployed on sepolia
+    address v3QuoteVerifier = address(0x6E64769A13617f528a2135692484B681Ee1a7169);
+
+    function setUp() public {
+        vm.createSelectFork("https://rpc.ankr.com/eth_sepolia");
+        // Get the instance of the DCAP Attestation QuoteVerifier on the Arbitrum Sepolia Rollup
+        vm.startPrank(adminTEE);
+
+        espressoSGXTEEVerifier =
+            new EspressoSGXTEEVerifier(enclaveHash, enclaveSigner, v3QuoteVerifier);
+        espressoTEEVerifier = new EspressoTEEVerifier(espressoSGXTEEVerifier);
+        vm.stopPrank();
+    }
+
+    function testRegisterSigner() public {
```

Comment:
I see that you have tests in the SGXVerifier test file for invalid quotes and such, but shouldn't we also test invalid registration patterns here? At the very least with an invalid tee type being passed in?

Reply:
okay will do!

Reply:
https://github.com/EspressoSystems/espresso-tee-contracts/pull/2/commits/a2bf9e83c0441bd275e4cb10f5bb23b8d440a8ad

Reply:
Hmm that's weird. I know the solidity compiler represents enums as `uint`'s. There isn't way to pass a `uint` to the function or like trick it by casting a `uint` into the enum type that would exist but isn't a currently named member of the enum? Strange :laughing:

Reply:
hmm let me try!